Automattic/wp-calypso
# Add fb and adwords tracking for akismet purchases

This diff adds FB and Adwords tracking for akismet purchases in the same way Jetpack purchases are tracked

P2: pbNhbs-5df-p2

## Proposed Changes

* Add new Akismet constants to the `calypso-products` package
* Identify Akismet products in cart splitting function in order to conditionally send tracking events
* Add tracking events for akismet products

## Testing Instructions

For reference when testing, these are the product slugs for Akismet products
* ak_plus_yearly_1
* ak_plus_monthly_1
* ak_plus_yearly_2
* ak_plus_monthly_2
* ak_plus_yearly_3
* ak_plus_monthly_3
* ak_plus_yearly_4
* ak_plus_monthly_4
* ak_ent_yearly_1
* ak_ent_monthly_1
* ak_ep350k_yearly_1
* ak_ep350k_monthly_1
* ak_ep2m_yearly_1
* ak_ep2m_monthly_1
* ak_epgt2m_yearly_1
* ak_epgt2m_monthly_1

1. Checkout this branch
2. Run `yarn start`
3. Point `public-api.wordpress.com` to your sandbox and enable store sandbox
4. Make sure you have a test site that you can attach the akismet license to in order to complete the checkout ( I used a personal wordpress site WITHOUT JETPACK, for some reason I could not get this purchase to go through on an atomic site with Jetpack )
5. Enable ad-tracking in your config (can be found here: fbhepr%2Skers%2Spnylcfb%2Spbasvt%2Sqrirybczrag.wfba%3Se%3Qr497735s%2329-og)
6. Go to `http://calypso.localhost:3000/checkout/{product_slug}` (for product_slug, see reference above)
7. You will have to pick a site to attach the license to. Once you are redirected to the cart, add `?flags=ad-tracking,cookie-banner` to the end of the URL and refresh
8. Ensure that you allow for advertising trackers
Run
`document.cookie = 'sensitive_pixel_options=%7B%22ok%22%3Atrue%2C%22buckets%22%3A%7B%22essential%22%3Atrue%2C%22analytics%22%3Atrue%2C%22advertising%22%3Atrue%7D%7D'`
in the console
9. Run `localStorage.setItem( 'debug', 'calypso:analytics:*' );` in the console
10. Go through with the purchase
11. Confirm you see an Adwords event similar to this ( I have not confirmed the full tracking ID for this, this is in progress )
![image](https://user-images.githubusercontent.com/65001528/219208261-beaab414-b219-446a-9931-66e0ee793517.png)
12. Confirm you see a Facebook event similar to this ( ensure the 2nd item in the object is tracking ID: 485349158311379 )
![image](https://user-images.githubusercontent.com/65001528/219208353-8d2323e4-049d-4ae1-9f14-0c3a48edb829.png)
13. Ensure the same events for Jetpack are NOT recorded in addition to these


## Pre-merge Checklist
